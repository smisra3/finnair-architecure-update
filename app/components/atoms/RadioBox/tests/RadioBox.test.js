import { shallow, mount } from 'enzyme';
import RadioBoxStyle, { RadioBox } from '../RadioBox';

const props = {
  onChecked: jest.fn(),
  onFocus: true,
};

describe('RadioBox Component with style ', () => {
  test('should render correctly with styles', () => {
    let checkBox = '';
    checkBox = shallow(<RadioBoxStyle />);
    expect(checkBox).toMatchSnapshot();
  });
});

describe('RadioBox Component ', () => {
  test('should render correctly', () => {
    let checkBox = '';
    checkBox = shallow(<RadioBox />);
    expect(checkBox).toMatchSnapshot();
  });
  test('should render checkbox checked', () => {
    const checkBoxChecked = shallow(<RadioBox isChecked />);
    expect(checkBoxChecked).toMatchSnapshot();
  });
  test('should render checkbox without label', () => {
    const checkBoxChecked = shallow(<RadioBox hideLabel />);
    expect(checkBoxChecked).toMatchSnapshot();
  });
  test('should trigger Toggle State', () => {
    const checkBoxChecked = shallow(<RadioBox hideLabel {...props} />);
    jest.spyOn(checkBoxChecked.instance(), 'toggleState');
    checkBoxChecked.find('input').simulate('change');
  });
  test('should trigger focus event', () => {
    const checkBoxChecked = shallow(<RadioBox hideLabel />);
    checkBoxChecked.find('input').simulate('focus');
    expect(checkBoxChecked.state().isFocused).toEqual(true);
  });

  test('should trigger blur State', () => {
    const checkBoxChecked = shallow(<RadioBox hideLabel />);
    checkBoxChecked.find('input').simulate('blur');
    expect(checkBoxChecked.state().isFocused).toEqual(false);
  });

  test('should mount', () => {
    const checkBoxChecked = mount(<RadioBox hideLabel />);
    checkBoxChecked.setState({ checked: true });
    checkBoxChecked.instance().componentWillReceiveProps({ isChecked: true });
    expect(checkBoxChecked).toMatchSnapshot();
  });

  test('should mount', () => {
    const checkBoxChecked = shallow(<RadioBox hideLabel />);
    checkBoxChecked.setState({ checked: true });
    checkBoxChecked.instance().componentWillReceiveProps({ isChecked: true });
    expect(checkBoxChecked).toMatchSnapshot();
  });
});
