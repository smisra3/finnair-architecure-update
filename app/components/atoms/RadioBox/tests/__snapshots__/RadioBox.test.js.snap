// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBox Component  should mount 1`] = `
<RadioBox
  disabled={false}
  hideLabel={true}
  imageUrl=""
  inheritedStyles=""
  isChecked={false}
  label={null}
  labelOnly={false}
  onBlur={[Function]}
  onFocus={[Function]}
>
  <label
    className="undefined is-checked"
    hideLabel={true}
    inheritedStyles=""
    label={null}
  >
    <input
      checked={false}
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="radio"
    />
    <span
      className="radio radio-tick radio-checked"
    />
  </label>
</RadioBox>
`;

exports[`RadioBox Component  should mount 2`] = `
<label
  className="undefined is-checked"
  hideLabel={true}
  inheritedStyles=""
  label={null}
>
  <input
    checked={false}
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <span
    className="radio radio-tick radio-checked"
  />
</label>
`;

exports[`RadioBox Component  should render checkbox checked 1`] = `
<label
  className="undefined is-checked"
  hideLabel={false}
  inheritedStyles=""
  label={null}
>
  <input
    checked={true}
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <span
    className="radio radio-tick radio-checked"
  />
</label>
`;

exports[`RadioBox Component  should render checkbox without label 1`] = `
<label
  className="undefined"
  hideLabel={true}
  inheritedStyles=""
  label={null}
>
  <input
    checked={false}
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <span
    className="radio radio-tick  radio-unchecked"
  />
</label>
`;

exports[`RadioBox Component  should render correctly 1`] = `
<label
  className="undefined"
  hideLabel={false}
  inheritedStyles=""
  label={null}
>
  <input
    checked={false}
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
  />
  <span
    className="radio radio-tick  radio-unchecked"
  />
</label>
`;

exports[`RadioBox Component with style  should render correctly with styles 1`] = `
<RadioBox
  className="sc-bdVaJa fbJpqK"
  disabled={false}
  hideLabel={false}
  imageUrl=""
  inheritedStyles=""
  isChecked={false}
  label={null}
  labelOnly={false}
  onBlur={[Function]}
  onFocus={[Function]}
/>
`;
